gulp/tasks/watch.js

```diff
@@ -1,13 +1,30 @@
 module.exports = function(gulp) {
-  gulp.task('watch', function() {
-    gulp.watch(gulp.config.source + '/styles/**/*.less', ['less']);
-    gulp.watch(gulp.config.source + '/js/**/*.js', ['browserify']);
-    gulp.watch([
-        gulp.config.source + '/*.html',
-        gulp.config.source + '/img/**/*.ico',
-        gulp.config.source + '/fonts/**/*',
-        gulp.config.source + '/videos/**/*'
-    ], ['copy']);
-    gulp.watch(gulp.config.source + '/views/**/*.jade', ['jade']);
+  var watch = require('gulp-watch');
+
+  gulp.task('watch', ['build'], function() {
+    watch(gulp.config.source + '/styles/**/*.less', function() {
+      gulp.start('less');
+    });
+
+    watch(gulp.config.source + '/js/**/*.js', function() {
+      gulp.start('browserify');
+    });
+
+    watch([
+      gulp.config.source + '/*.html',
+      gulp.config.source + '/img/**/*.{png,gif,jpg,jpeg,svg}',
+      gulp.config.source + '/fonts/**/*',
+      gulp.config.source + '/videos/**/*'
+    ], function() {
+      gulp.start('copy');
+    });
+
+    watch(gulp.config.source + '/views/**/*.jade', function() {
+      gulp.start('jade');
+    });
+
+    watch(gulp.config.source + '/img/**/*.{png,gif,jpg,jpeg,svg}', function() {
+      gulp.start('images');
+    });
```

Comment:
Why use `gulp-watch` instead of `gulp.watch`? Seems like we're not even using it's signature streaming feature.